src/Body.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Allows special input as body
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+interface Body
+{
+    /**
+     * Returns a set of headers which is needed to correctly send the body
+     *
+     * Note: these headers get overwritten by headers manually passed to the client
+     *
+     * Content-Length is calculated automatically if possible
+     *
+     * @return array
+     */
+    public function getContentHeaders();
+
+    /**
+     * Convert data to a format which can be used to create a proper PSR-7 Stream
+     *
+     * @return string|StreamInterface
```

Comment:
We have a custom MessageFactory interface which is recommended to be used. This ensures you can use whatever PSR7 implementation in your clients. There is already some kind of documentation about it, for reference check the docs repo.

Presonally I think that the diactoros behavior is a bit weird. I think using string as a simple body is much more logical than using it as a URL to a stream.